### PR TITLE
ci: update pip requirements.txt

### PR DESCRIPTION
**Pull request by thst-nordic**
* adding pylint for compliance.py
* generate requirements-ci.txt from multiple projects

Signed-off-by: Thomas Stilwell <Thomas.Stilwell@nordicsemi.no>

### CODE REVIEW

File: scripts/requirements-ci.sh [hunk 1]
```diff
@@ -29,3 +29,7 @@ pip3 freeze > $OUT_FILE
 
 deactivate
 rmvirtualenv pip-freeze-ci-venv
+
+
+# Sphinx==2.4.4
+# breathe==4.14.2
```
Comment (tejlmand) [inline scripts/requirements-ci.sh:35]: Just curious.
Why add those lines as they are anyway commented out ?
Also, the syntax used here seems wrong for a bash script.

File: scripts/requirements-ci.txt [hunk 2]
```diff
@@ -2,55 +2,56 @@ alabaster==0.7.12
...
 cmsis-pack-manager==0.2.10
 colorama==0.4.3
 CommonMark==0.5.4
 configobj==5.0.6
-cryptography==2.9
+cryptography==2.9.2
 docopt==0.6.2
 docutils==0.16
 ecdsa==0.15
-editdistance==0.5.3
```
Comment (tejlmand) [inline scripts/requirements-ci.txt:21]: you don't need this anymore, or are you sourcing `scripts/requirements-extra.txt` as well ?
Please note, the version in `scripts/requirements-extra.txt` is only `0.5.0`

File: scripts/requirements.txt [hunk 3]
```diff
@@ -4,3 +4,4 @@ intelhex
 recommonmark==0.4.0
 sphinxcontrib-mscgen
 west>=0.7.2
+pylint
```
Comment (tejlmand) [inline scripts/requirements.txt:7]: why is this a requirement for all users, and not just ci ?

File: scripts/requirements-ci.txt (hunk 2 above)
Comment (thst-nordic) [reply scripts/requirements-ci.txt:21]: ci doesn't need the packages in extra.txt, they were added by mistake last release

File: scripts/requirements.txt (hunk 3 above)
Comment (thst-nordic) [reply scripts/requirements.txt:7]: req-ci.txt is generated from this file.

File: scripts/requirements-ci.sh (hunk 1 above)
Comment (thst-nordic) [reply scripts/requirements-ci.sh:35]: I can remove them.
what's wrong with the syntax?

Comment (tejlmand) [reply scripts/requirements-ci.sh]: That doing `breathe==4.14.2` in bash would not work (as if you removed the `#`), it seems to be the syntax you would use in a pip requirements.txt file 😉

File: scripts/requirements-ci.txt (hunk 2 above)
Comment (tejlmand) [reply scripts/requirements-ci.txt:21]: thanks for explaining

File: scripts/requirements.txt (hunk 3 above)
Comment (tejlmand) [reply scripts/requirements.txt:7]: got it.

**Pull request merged**